src/EditorFeatures/Core/Implementation/Formatting/InferredIndentationDocumentOptionsProviderFactory.cs

```diff
@@ -38,8 +38,21 @@ private class DocumentOptionsProvider : IDocumentOptionsProvider
             public DocumentOptionsProvider(IIndentationManagerService indentationManagerService)
                 => _indentationManagerService = indentationManagerService;
 
-            public Task<IDocumentOptions?> GetOptionsForDocumentAsync(Document document, CancellationToken cancellationToken)
-                => Task.FromResult<IDocumentOptions?>(new DocumentOptions(document.Project.Solution.Workspace, document.Id, _indentationManagerService));
+            public async Task<IDocumentOptions?> GetOptionsForDocumentAsync(Document document, CancellationToken cancellationToken)
+            {
+                // Languages such as Razor can specify their own formatting options.
```

Comment:
So it's a bit odd that we are having to modify the InferredIndentationDocumentOptionsProviderFactory here when we could have a new MEF export. The Razor ExternalAccess could directly export it's own IDocumentOptionsProviderFactory that imports the Razor service directly. This would mean you can keep almost the entire change in the Razor layer and not have to introduce the extra layer in the middle: just rename IDocumentOptionsService to IRazorDocumentOptionsService and you can get away with bit less wrapping.

Reply:
(it's also otherwise odd we'd be having this at an EditorFeatures layer, since what happens for VS Code?)